### ISSUE TITLE
Refactor System.Reflection.MetadataLoadContext tests to work for single file

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#72908]
MetadataLoadContext test has a clever trick to map runtime types to their MetadataLoadContext version by using Assembly.Location, opening them in a metadata context and finding the corresponding type. This doesn't work in single-file testing in general and NativeAOT in particular.

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Reflection.MetadataLoadContext/tests/src/TestUtils/TestUtils.JittedRuntimes.cs#L11-L26

The best way forward would probably be to introduce a test assembly that will have all the types we need to project and to include it as Content. Like so:

https://github.com/dotnet/runtime/blob/8abc2519948afb07ecb41489a012d0546d15aa17/src/libraries/System.Diagnostics.FileVersionInfo/tests/System.Diagnostics.FileVersionInfo.Tests/System.Diagnostics.FileVersionInfo.Tests.csproj#L46-L49

We can have the test types in both the test project and in the Content assembly - the projection would be updated to look in the Content assembly instead of Assembly.Location.

We'll probably need another Content assembly that will be a simulated CoreLib with whatever dummy content we need to make the test assembly build. It wouldn't be the first fake CoreLib in the runtime repo.

**Comment by steveharter** [comment]
Linking https://github.com/dotnet/runtime/pull/72917 and https://github.com/dotnet/runtime/issues/27803

**Comment by steveharter** [comment]
Moving to v9 based on priority+resources.

**Comment by steveharter** [comment]
@MichalStrehovsky is this still a valid scenario and is the possible implementation below still valid?

> The best way forward would probably be to introduce a test assembly that will have all the types we need to project and to include it as Content.

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky is this still a valid scenario and is the possible implementation below still valid?
> 
> > The best way forward would probably be to introduce a test assembly that will have all the types we need to project and to include it as Content.

Yes, it's still a test gap and including as Content should work. E.g. the System.Diagnostics.FileVersionInfo.Tests test does that to include System.Diagnostics.FileVersionInfo.TestAssembly (a test assembly file that is loaded to check its version).

**Comment by ericstj** [comment]
> We'll probably need another Content assembly that will be a simulated CoreLib with whatever dummy content we need to make the test assembly build. It wouldn't be the first fake CoreLib in the runtime repo.

What about just including `System.Runtime.dll` with the test assembly.  @jkotas and I were discussing improving the test coverage with more behavior diffing in https://github.com/dotnet/runtime/pull/116080#discussion_r2116717994 and we have other reasons to include it.

So we could build a test assembly of interesting types, and include it as content, along with the minimal set of references it required to build.

I guess including the test assembly itself as content could also solve this, but I like the better factoring of having an assembly of only interesting test types.

**Comment by jkotas** [comment]
> What about just including System.Runtime.dll

Yes, including System.Runtime ref assembly should be enough.
